autosklearn/ensemble_builder.py

```diff
@@ -526,14 +526,7 @@ def get_n_best_preds(self):
 
         for k, _, _ in sorted_keys[:keep_nbest]:
             if self.read_preds[k][Y_ENSEMBLE] is None:
-                if k.endswith("gz"):
-                    open_method = gzip.open
-                elif k.endswith("npy"):
-                    open_method = open
-                else:
-                    raise ValueError("Unknown filetype %s" % k)
-                with open_method(k, 'rb') as fp:
-                    self.read_preds[k][Y_ENSEMBLE] = self._read_np_fn(fp=fp)
+                self.read_preds[k][Y_ENSEMBLE] = self._read_np_fn(k)
```

Comment:
I'm wondering whether this loading logic should go after the part which contains '#consider performance_range_threshold'?

Reply:
Or even better after dropping models/loaded predictions. I guess that would be maximally memory efficient.